### PR TITLE
It contains integration with service spec builder and service creation.

### PR DESCRIPTION
**Pull request by jakhani**

### CODE REVIEW

File: azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java [hunk 1]
```diff
@@ -334,7 +348,30 @@ private void populateJobTypeForFlow(final ExecutableNode node, Set<String> jobTy
...
+          .withExecId(String.valueOf(executionId))
+          .withServiceName(getServiceName(executionId))
+          .withNamespace(this.namespace)
+          .withApiVersion(SERVICE_API_VERSION_2)
+          .withKind(MAPPING)
+          .withPort(String.valueOf(this.servicePort))
+          .withTimeoutMs(String.valueOf(this.serviceTimeout))
+          .build();
+    } catch (final IOException e) {
+      logger.error("Unable to create service object as service spec is missing: " + e.getMessage());
```
Comment (sshardool) [inline azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java:366]: Nit:
The error message seems quite specific to the working of AzKubernetesV1ServiceBuilder  for IOException ?
Maybe keep the error message generic or alternatively convert the IOException to something more generic within  AzKubernetesV1ServiceBuilder.build() ?

**Review by sshardool — DISMISSED** [review]
Looks good apart from a nit. Please feel to address in a subsequent pr.

**Review by arvindpruthi — DISMISSED** [review]

File: azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java (hunk 1 above)
Comment (jakhani) [reply azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java]: Good point. Fixed it.

**Review by sshardool — APPROVED** [review]
Thanks for addressing this!

**Review by arvindpruthi — APPROVED** [review]

**Pull request merged**